CppSamples/Layers/AddVectorTiledLayerFromCustomStyle/AddVectorTiledLayerFromCustomStyle.cpp

```diff
@@ -0,0 +1,216 @@
+{
+  if (!mapView || mapView == m_mapView)
+  {
+    return;
+  }
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  const Viewpoint defaultViewpoint(10, 5.5, 1e8);
+  m_map->setInitialViewpoint(defaultViewpoint);
+
+  populatePortalIdMap();
+  emit mapViewChanged();
+}
+
+void AddVectorTiledLayerFromCustomStyle::populatePortalIdMap()
+{
+  // The labels and portal item IDs of the online styles.
+  m_portalIds.insert("Default", "1349bfa0ed08485d8a92c442a3850b06");
+  m_portalIds.insert("Style 1", "bd8ac41667014d98b933e97713ba8377");
+  m_portalIds.insert("Style 2", "02f85ec376084c508b9c8e5a311724fa");
+  m_portalIds.insert("Style 3", "1bf0cc4a4380468fbbff107e100f65a5");
+
+  // The labels and portal item IDs of the offline styles.
+  m_portalIds.insert("Offline custom style - Light", "e01262ef2a4f4d91897d9bbd3a9b1075");
+  m_portalIds.insert("Offline custom style - Dark", "ce8a34e5d4ca4fa193a097511daa8855");
+
+  initialize();
+
+  m_styleNames = m_portalIds.keys();
+  emit setStyleNames();
+}
+
+void AddVectorTiledLayerFromCustomStyle::initialize()
+{
+  Portal* portal = new Portal(this);
+
+  // Store a list all portal items.
+  for (const QString& itemID : m_portalIds.values())
+  {
+    PortalItem* portalItem = new PortalItem(portal, itemID, this);
+    m_vectorTiledLayers.insert(m_portalIds.key(itemID), portalItem);
+  }
+
+  // Export offline custom styles.
+  connect(this, &AddVectorTiledLayerFromCustomStyle::styleExported, this, [this](const QString& styleName, ItemResourceCache* cache)
+  {
+    if (styleName == "Light")
+    {
+      m_lightStyleResourceCache = cache;
+    }
+    else if (styleName == "Dark")
+    {
+      m_darkStyleResourceCache = cache;
+    }
+  });
+
+  exportStyle(m_vectorTiledLayers.value("Offline custom style - Light"), "Light");
+  exportStyle(m_vectorTiledLayers.value("Offline custom style - Dark"), "Dark");
+}
+
+void AddVectorTiledLayerFromCustomStyle::selectionChanged(const QString style)
+{
+  // Check if the user selected a offline custom style.
+  // Create a new basemap with the appropriate style.
+  if (style.contains("Offline"))
+  {
+    ItemResourceCache* cache = nullptr;
+
+    if (style.contains("Light"))
+    {
+      cache = m_lightStyleResourceCache;
+    }
+    else
+    {
+      cache = m_darkStyleResourceCache;
+    }
+
+    m_map->setBasemap(new Basemap(new ArcGISVectorTiledLayer(new VectorTileCache(m_dataPath), cache)));
+
+    const Viewpoint dodgeCityViewpoint(37.76528, -100.01766, 4e4);
+    m_mapView->setViewpointAsync(dodgeCityViewpoint);
+  }
+  else
+  {
+    m_map->setBasemap(new Basemap(new ArcGISVectorTiledLayer(m_vectorTiledLayers.value(style))));
+    const Viewpoint defaultViewpoint(10, 5.5, 1e8);
+    m_mapView->setViewpointAsync(defaultViewpoint);
+  }
+}
+
+void AddVectorTiledLayerFromCustomStyle::exportStyle(PortalItem* vectorTiledLayer, const QString& styleName)
+{
+  vectorTiledLayer->load();
```

Comment:
the load should be called after the connection is setup. There's a slim chance that load could complete before the connections are setup